docs/UniversalLinking.md

```diff
@@ -0,0 +1,54 @@
+# Universal Linking
+
+Universal links are an elegant solution for deep linking in a React Native project. Refer to the [React Native doc on Linking](https://reactnative.dev/docs/linking) for details.
+
+## iOS Considerations
+
+React Native apps supporting universal linking typically use [this snippet](https://reactnative.dev/docs/linking#enabling-deep-links) which calls `RCTLinkingManager`:
+
+```objc
+- (BOOL)application:(UIApplication *)application continueUserActivity:(nonnull NSUserActivity *)userActivity
+ restorationHandler:(nonnull void (^)(NSArray<id<UIUserActivityRestoring>> * _Nullable))restorationHandler
+{
+ return [RCTLinkingManager application:application
+                  continueUserActivity:userActivity
+                    restorationHandler:restorationHandler];
+}
+```
+
+Implementing the `UIApplicationDelegate.application(_:continue:restorationHandler:)` function like above causes all links (deep links _and_ regular web links) from the Appcues SDK to be passed to your implementation of `Linking.addEventListener('url', callback)`. If your `Linking` callback is not prepared for non-deep links, the result will be button links in Appcues flows being broken.
+
+To resolve this, you may define a custom key, `AppcuesUniversalLinkHostAllowList`, in your _Info.plist_ file with an array of (string) host values to control which URLs are passed to `RCTLinkingManager` to be handled as a univeral link. A URL whose host matches any of the provided host values will be treated as a universal link and passed to `Linking.addEventListener('url', callback)`. Non matching URLs will be handled by the Appcues SDK.
+
+If the `AppcuesUniversalLinkHostAllowList` key is absent from _Info.plist_, all https links will be treated as potential universal links.
+
+Example _Info.plist_:
```

Comment:
only thought is maybe add a simple example usage to this section, i.e. saying, if you use the Info.plist shown above,

https://www.my-domain.com/home --> this WILL be sent to your app to handle as a deep link

https://www.appcues.com/mobile -> the Appcues SDK will handle this one, and open in the browser (or in-app browser) as configured in the mobile builder

Reply:
Good idea. Just to note `www.my-domain.com` wouldn't be covered by `my-domain.com`, so I added a specific subdomain example too.

Reply:
oh right, cool thank you